unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Texture/AssetPromise_Texture.cs

```diff
@@ -130,20 +126,30 @@ protected override bool AddToLibrary()
             return true;
         }
 
-        string ConstructId(string textureUrl, TextureWrapMode textureWrapMode, FilterMode textureFilterMode)
-        {
-            return ((int)textureWrapMode).ToString() + ((int)textureFilterMode).ToString() + textureUrl;
-        }
+        string ConstructId(string textureUrl, TextureWrapMode textureWrapMode, FilterMode textureFilterMode) { return ((int)textureWrapMode).ToString() + ((int)textureFilterMode).ToString() + textureUrl; }
 
         public override object GetId()
         {
             // We only use the id-with-settings when storing/reading from the library
             return idWithDefaultTexSettings;
         }
 
-        public bool UsesDefaultWrapAndFilterMode()
+        public bool UsesDefaultWrapAndFilterMode() { return wrapMode == DEFAULT_WRAP_MODE && filterMode == DEFAULT_FILTER_MODE; }
+
+        static byte[] questionMarkPNG = { 137, 80, 78, 71, 13, 10, 26, 10, 0, 0, 0, 13, 73, 72, 68, 82, 0, 0, 0, 8, 0, 0, 0, 8, 8, 2, 0, 0, 0, 75, 109, 41, 220, 0, 0, 0, 65, 73, 68, 65, 84, 8, 29, 85, 142, 81, 10, 0, 48, 8, 66, 107, 236, 254, 87, 110, 106, 35, 172, 143, 74, 243, 65, 89, 85, 129, 202, 100, 239, 146, 115, 184, 183, 11, 109, 33, 29, 126, 114, 141, 75, 213, 65, 44, 131, 70, 24, 97, 46, 50, 34, 72, 25, 39, 181, 9, 251, 205, 14, 10, 78, 123, 43, 35, 17, 17, 228, 109, 164, 219, 0, 0, 0, 0, 73, 69, 78, 68, 174, 66, 96, 130, };
+        internal bool IsQuestionMarkPNG(Texture tex)
         {
-            return wrapMode == DEFAULT_WRAP_MODE && filterMode == DEFAULT_FILTER_MODE;
+            if (!tex)
+                return true;
+
+            if (tex.width * tex.height != questionMarkPNG.Length )
+                return false;
+
+            byte[] png1 = (tex as Texture2D).EncodeToPNG();
```

Comment:
Not sure about unity, but encoding textures back to PNG requires querying the GPU memory which is a pipeline blocking operation. I'd try to avoid this as much as possible. Do we have the underlying number of updates to a texture or something else in the texture loader we can hook into?

Reply:
I think you are right here, therefore the early return. As commented in your previous concern, I'm gonna solve it comparing against the texture's width and height.

So we would end up doing this check only against other 8x8 textures and that should be affordable.

Reply:
Not sure how much performance we would gain (it could be meassured with [`System.Diagnostics.Stopwatch`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch)) but if we want to avoid the PNG encoding altogether, we may be able to compare using the Tex2D pixels as we do [here](https://github.com/decentraland/explorer/blob/e11f86b07b114ae230ef01eb2d17dd7726b16e4e/unity-client/Assets/Scripts/Tests/VisualTests/VisualTestHelpers.cs#L208) for our visual tests.

Reply:
First we would need to have the texture itself contained in the project and we could run into false-negative issues due to floating point precission errors.

That's doable as well but I don think we are gaining that much.